rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/SystemProperty.java

```diff
@@ -26,6 +26,7 @@
  */
 public abstract class SystemProperty {
     // KEYS
+    public static final String GRAPH_LIBRARY = "gaffer.graph.library.path";
```

Comment:
This should be:

public static final String GRAPH_LIBRARY_CLASS = "gaffer.graph.library.class";
public static final String GRAPH_LIBRARY_CONFIG = "gaffer.graph.library.config";

Then users can choose what graph library they want to use.
We could then load it using something like:

```java
String graphLibraryClassName = System.getProperty(SystemProperty.GRAPH_LIBRARY_CLASS);
if(null != graphLibraryClassName) {
    GraphLibrary library = null;
    try {
        library = Class.forName(graphLibraryClassName).asSubClass(GraphLibrary.class).newInstance();
    } catch (ClassNotFoundException e) {
        throw new RuntimeException("Unknown GraphLibrary class: + " + graphLibraryClassName);
    }
    library.initialise(System.getProperty(SystemProperty.GRAPH_LIBRARY_CONFIG));
}
```